src/test/java/net/oauth/jsontoken/JsonTokenTest.java

```diff
@@ -1,28 +1,126 @@
+import org.joda.time.Instant;
+
+import java.security.SignatureException;
 
 public class JsonTokenTest extends JsonTokenTestBase {
 
-  public void testCreateJsonToken() throws Exception {
+  private static final String TOKEN_STRING_NULL_FIELDS = "eyJhbGciOiJIUzI1NiJ9.eyJoZWxsbyI6IndvcmxkIn0.URnYdSXdAAEukebqZgMq6oFjK4E9cEZlfvO8tBe_WeA";
+  private static final String TOKEN_STRING_EMPTY_PAYLOAD = "eyJhbGciOiJIUzI1NiJ9.e30.XmNK3GpH3Ys_7wsYBfq4C3M6goz71I7dTgUkuIa5lyQ";
+
+  public void testSignAndSerialize() throws Exception {
     HmacSHA256Signer signer = new HmacSHA256Signer("google.com", "key2", SYMMETRIC_KEY);
 
     JsonToken token = new JsonToken(signer, clock);
     token.setParam("bar", 15);
     token.setParam("foo", "some value");
     token.setAudience("http://www.google.com");
     token.setIssuedAt(clock.now());
-    token.setExpiration(clock.now().withDurationAdded(60, 1));
+    token.setExpiration(clock.now().plus(Duration.standardSeconds(1)));
 
     assertEquals(TOKEN_STRING, token.serializeAndSign());
   }
 
-  public void testCreateAnotherJsonToken() throws Exception {
+  public void testSignAndSerializeNullFields() throws Exception {
     HmacSHA256Signer signer = new HmacSHA256Signer(null, (String) null, "secret".getBytes());
 
     JsonToken token = new JsonToken(signer, clock);
     token.setParam("hello", "world");
-    String encodedToken = token.serializeAndSign();
+
+    assertEquals(TOKEN_STRING_NULL_FIELDS, token.serializeAndSign());
+  }
+
+  public void testSignAndSerializeEmptyPayload() throws Exception {
+    HmacSHA256Signer signer = new HmacSHA256Signer(null, (String) null, "secret".getBytes());
+    JsonToken token = new JsonToken(signer, clock);
+    assertEquals(TOKEN_STRING_EMPTY_PAYLOAD, token.serializeAndSign());
+  }
+
+  private JsonObject getFullHeader() {
+    JsonObject header = new JsonObject();
+    header.addProperty(JsonToken.ALGORITHM_HEADER, "HS256");
+    header.addProperty(JsonToken.KEY_ID_HEADER, "key2");
+    return header;
+  }
+
+  private JsonObject getFullPayload() {
+    JsonObject payload = new JsonObject();
+    payload.addProperty(JsonToken.ISSUER, "google.com");
+    payload.addProperty("bar", 15);
+    payload.addProperty("foo", "some value");
+    payload.addProperty(JsonToken.AUDIENCE, "http://www.google.com");
+    payload.addProperty(JsonToken.ISSUED_AT, 1276669722);
+    payload.addProperty(JsonToken.EXPIRATION, 1276669723);
+    return payload;
+  }
+
+  private void fullPayloadAssertions(JsonToken token) {
+    assertEquals("google.com", token.getIssuer());
+    assertEquals("http://www.google.com", token.getAudience());
+    assertEquals(new Instant(1276669722000L), token.getIssuedAt());
+    assertEquals(new Instant(1276669723000L), token.getExpiration());
+    assertEquals(15, token.getParamAsPrimitive("bar").getAsLong());
+    assertEquals("some value", token.getParamAsPrimitive("foo").getAsString());
+  }
+
+  public void testParse() throws Exception {
```

Comment:
nit: I prefer testConstructFromJson here and below, because parse usually is referring to taking the 3 part base64 encoded jwt string and reading the data from it.

Reply:
Changed here and below